### PR TITLE
Docs improvements

### PR DESCRIPTION
**Pull request by TinaTiel**
- Reorganized README, adding subsections etc.
- Moved transfer noticed to CONTRIBUTING
- Updated gradle calls in CONTRIBUTING

I recommend viewing in your browser, in https://github.com/obs-websocket-community-projects/obs-websocket-java/tree/docs-improvements

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,125 +1,126 @@
...
 
-Take a look at the [**example**](example/src/main/java/io/obswebsocket/community/client/example/Example.java)
+### Error Handling
+When the OBS Websocket encounters a significant problem (failed authentication, unknown message type, etc) it will close the connection. These events can be observed in the  `onClose` lifecycle event. 
 
-Examples can be found [**here**](client/src/test/java/net/twasi/obsremotejava/test/OBSRemoteControllerTest.java). Just uncomment the requests you want to test or copy.
+Individual requests will include a status code(SourceNotFound, HotkeyNotFound, etc) as part of the response and do not result in the connection being closed.
 
----
+The `onError` communicator and controller lifecycle events are reserved for exceptional events (null pointers and other development errors) and for failure to connect to OBS (for example, if OBS Websocket isn't installed, if OBS isn't running, or it isn't accessible over the network).
```
Comment (ChristopheCVB) [inline README.md:123]: `... and other exceptions`?

File: README.md [hunk 2]
```diff
@@ -131,9 +132,10 @@ configure Maven to use Logback instead:
 ```
```
Comment (ChristopheCVB) [inline README.md:132]: xml

File: README.md [hunk 3]
```diff
@@ -152,38 +154,55 @@ configure Maven to use Logback instead:
 </dependencies>
 ```
```
Comment (ChristopheCVB) [inline README.md:155]: Provide the gradle too?

File: README.md [hunk 4]
```diff
@@ -152,38 +154,55 @@ configure Maven to use Logback instead:
...
 
-## Contributing / Issues
+## Migrating from Twasi 1.X.X
+Versions of this library prior to 2.0.0 were intended for OBS Websockets 4.X.X, and are not backwards compatible.
```
Comment (ChristopheCVB) [inline README.md:158]: `and are not backwards compatible.` -> `and 2.0.0 is not backwards compatible.`

**Review by ChristopheCVB — CHANGES_REQUESTED** [review]
Astounding doc! Just some minor changes/suggestions requested

**Review by Pjiesco — DISMISSED** [review]

File: README.md (hunk 3 above)
Comment (ChristopheCVB) [reply README.md:154]:
```groovy
dependencies {
    implementation('io.obs-websocket.community:client:...') {
        exclude group: 'org.slf4j', module: 'slf4j-simple'
    }
    implementation('ch.qos.logback:logback-classic:1.1.7')
}
```

File: README.md [hunk 5]
```diff
@@ -1,125 +1,125 @@
 # OBS WebSocket Java
 
 <p align="center">
-  <img src="/.github/images/obs-ws-java.png" width=150 align="center">
+  <img src=".github/images/obs-ws-java.png" width=150 align="center">
+  <p align="center">A java library for the <a href="https://github.com/Palakis/obs-websocket">OBS-Studio websocket plugin</a> by <a href="https://github.com/Palakis">Palakis</a>).</p>
```
Comment (ChristopheCVB) [inline README.md:5]: remove `)`

**Review by ChristopheCVB — APPROVED** [review]

**Comment by TinaTiel** [comment]
Thanks for your feedback on this! Hopefully the docs are helpful to devs using this library :D

**Pull request merged**